rfi_file_monitor/utils/decorators.py

```diff
@@ -187,3 +188,14 @@ def wrapper(self: Operation, file: File):
             raise NotImplementedError(f"{type(file)} is currently unsupported")
 
     return wrapper
+
+def do_bulk_upload( process_existing_files: Callable[List]):
+    @functools.wraps(process_existing_files)
+    def wrapper(self: Engine, existing_files: List):
+        no_files = sum(len(fs) for _, _, fs, in os.walk(*args))
+        if no_files > 5000: # do not like this hard coded value but will put this in for performance testing
+            chunked_input = [existing_files[i:i + 5000] for i in range(0, len(existing_files), 5000)]
+            for rv in chunked_input:
+                process_existing_files(self, rv)
+                sleep(60) # calculate this based on the file weight?
```

Comment:
we want to update the weight dynamically without having to monitor the queue in the decorator